MekHQ/src/mekhq/gui/utilities/StaticChecks.java

```diff
@@ -405,6 +405,42 @@ public static boolean doAllUnitsHaveC3Master(Vector<Unit> units) {
         return true;
 
     }
+    
+    /**
+     * Used to test a selection of Units provided by the player and determine whether they all share a designated unitType. 
+     * @param units Vector of units that the player has selected
+     * @return false if any unit in the passed-in Vector does not have the specified unit type
+     */
+    public static boolean areAllUnitsSameType(Vector<Unit> units, int unitType) {
+        // For our purposes, a selection of tanks isn't the same unless they're all the same weight class
+        // Set this based on the first unit in the selection
+        double firstUnitWeight = units.get(0).getEntity() != null ? units.get(0).getEntity().getWeight() : 0;
+        boolean light = firstUnitWeight > 0 && firstUnitWeight <= 50;
```

Comment:
Is the > 0 necessary? Won't they all have at least some weight (if you have some unit).

Reply:
Seemed like a good idea, in case we managed to get a botched unit with negative or 0 weight